sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -134,8 +142,13 @@ cdef class HistogramBuilder:
                 shape=(self.n_features, self.n_bins),
                 dtype=HISTOGRAM_DTYPE
             )
+            bint has_interaction_cst = False
             int n_threads = self.n_threads
 
+        has_interaction_cst = allowed_features is not None
+        if has_interaction_cst:
+            n_allowed_features = allowed_features.shape[0]
+
         with nogil:
```

Comment:
This allows reusing thread, eventually reducing OpenMP's overhead.
```suggestion
        with nogil, parallel(num_threads=n_threads:
```

Reply:
Could we try this out in another PR?